results/cmd/api/main.go

```diff
@@ -170,16 +180,67 @@ func (s *server) DeleteTaskRunResult(ctx context.Context, req *pb.DeleteTaskRunR
 	return nil, nil
 }
 
+// ListTaskRunsResult receives a ListTaskRunRequest from users and return to users a list of TaskRuns according to the query
+func (s *server) ListTaskRunsResult(ctx context.Context, req *pb.ListTaskRunsRequest) (*pb.ListTaskRunsResponse, error) {
+	// get all taskruns from database
+	rows, err := s.db.Query("SELECT taskrunlog FROM taskrun")
+	if err != nil {
+		log.Printf("failed to query on database: %v", err)
+		return nil, status.Errorf(codes.NotFound, "No TaskRuns found for the query string due to internal server error, please try to enter the query again")
+	}
+	var taskRunList []*pb.TaskRun
+	for rows.Next() {
+		taskrun := &pb.TaskRun{}
+		var taskrunblob []byte
+		rows.Scan(&taskrunblob)
+		if err := proto.Unmarshal(taskrunblob, taskrun); err != nil {
+			log.Fatal("unmarshaling error: ", err)
+			return nil, status.Errorf(codes.NotFound, "No TaskRuns found for the query string due to internal server error, please try to enter the query again")
+
+		}
+		taskRunList = append(taskRunList, taskrun)
+	}
+	// return all taskruns back to users if empty query is given
+	if req.GetFilter() == "" {
+		return &pb.ListTaskRunsResponse{Items: taskRunList}, nil
+	}
+	ast, issues := s.env.Compile(req.GetFilter())
+	if issues != nil && issues.Err() != nil {
+		log.Printf("type-check error: %s", issues.Err())
+		return nil, status.Errorf(codes.NotFound, "No TaskRuns found for the query string due to wrong type in the query, please try to enter a query with correct type again")
```

Comment:
codes.InvalidArgument since this error happens if the users sends the wrong type?

Reply:
done